pyAKI/kdigo.py

```diff
@@ -12,10 +12,38 @@
     CreatininePreProcessor,
     DemographicsPreProcessor,
 )
-from utils import Dataset, DatasetType
+from utils import Dataset
 
 
 class Analyser:
+    """
+    Class for data analysis using probes and preprocessors.
+
+    This class provides functionality for analyzing data using a collection of probes and preprocessors.
+    It processes the input data through the specified preprocessors and applies the probes to perform
+    the analysis. The analysis results are returned as a DataFrame.
+
+    Args:
+        data (list[Dataset]): A list of Dataset objects containing the input data.
+        probes (list[Probe], optional): A list of Probe objects representing the analysis probes to apply.
```

Comment:
Maybe we should use [Optinal](https://docs.python.org/3/library/typing.html#typing.Optional) as type hints?